src/Components/WebAssembly/testassets/HostedInAspNet.Server/Startup.cs

```diff
@@ -39,13 +39,15 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, BootReso
                 app.UseWebAssemblyDebugging();
             }
 
+            app.UseBlazorFrameworkFiles("/app");
             app.UseBlazorFrameworkFiles();
             app.UseStaticFiles();
 
             app.UseRouting();
 
             app.UseEndpoints(endpoints =>
             {
+                endpoints.MapFallbackToFile("/app/{**slug:nonfile}", "app/index.html");
```

Comment:
Re-using the `HostedInAspNet.Server` app to host a second wasm app this way is a bit strange. It has a risk that the E2E test could pass falsely by loading assets from the root instead of the expected non-root pathbase.

A safer alternative would be for your E2E test to use `AspNetSiteServerFixture`'s `AdditionalArguments` property to pass a flag into `HostedInAspNet.Server`'s `Program` class saying that it wants to serve the `CustomPathBaseApp` app **instead** of `HostedInAspNet.Client` (not both at once).

Alternatively you could treat `CustomPathBaseApp` as a standalone app. The E2E test for it would need to inherit from `DevHostServerFixture<CustomPathBaseApp.Program>` and then can set its own `PathBase` in its constructor.

Reply:
@SteveSandersonMS let me see if I can include the flag separately, what we care about most is the hosted scenario. For standalone it doesn't really make sense.

Reply:
Ok, I've updated this to consider the original test, a test where we only map the app on the subpath and a final test where we map both apps.